stream_srtp.go

```diff
@@ -5,77 +5,89 @@ import (
 	"sync"
 
 	"github.com/pions/rtp"
+	"github.com/pions/transport/packetio"
 )
 
-type readResultSRTP struct {
-	len    int
-	header *rtp.Header
-}
-
 // ReadStreamSRTP handles decryption for a single RTP SSRC
 type ReadStreamSRTP struct {
 	mu sync.Mutex
 
 	isInited bool
 	isClosed chan bool
 
-	session   *SessionSRTP
-	ssrc      uint32
-	readCh    chan []byte
-	readRetCh chan readResultSRTP
+	session *SessionSRTP
+	ssrc    uint32
+
+	buffer *packetio.Buffer
 }
 
 // Used by getOrCreateReadStream
 func newReadStreamSRTP() readStream {
 	return &ReadStreamSRTP{}
 }
 
-// ReadRTP reads and decrypts full RTP packet and its header from the nextConn
-func (r *ReadStreamSRTP) ReadRTP(payload []byte) (int, *rtp.Header, error) {
-	select {
-	case <-r.session.closed:
-		return 0, nil, fmt.Errorf("SRTP session is closed")
-	case r.readCh <- payload:
-	case <-r.isClosed:
-		return 0, nil, fmt.Errorf("SRTP read stream is closed")
+func (r *ReadStreamSRTP) init(child streamSession, ssrc uint32) error {
+	sessionSRTP, ok := child.(*SessionSRTP)
+
+	r.mu.Lock()
+	defer r.mu.Unlock()
+
+	if !ok {
+		return fmt.Errorf("ReadStreamSRTP init failed type assertion")
+	} else if r.isInited {
+		return fmt.Errorf("ReadStreamSRTP has already been inited")
 	}
 
-	select {
-	case <-r.session.closed:
-		return 0, nil, fmt.Errorf("SRTP session is closed")
-	case res, ok := <-r.readRetCh:
-		if !ok {
-			return 0, nil, fmt.Errorf("SRTP read stream is closed")
-		}
-		return res.len, res.header, nil
+	r.session = sessionSRTP
+	r.ssrc = ssrc
+	r.isInited = true
+	r.isClosed = make(chan bool)
+
+	// Create a buffer with a 1MB limit
+	r.buffer = packetio.NewBuffer()
+	r.buffer.SetLimitSize(1000 * 1000)
```

Comment:
Can you make this a constant and pull it out of the function (just for visibility)